### PR TITLE
MODUSERBL-51: Implement endpoint for create/reset password link validation

### PR DESCRIPTION
**Pull request by opochepam**
[Jira link](https://issues.folio.org/browse/MODUSERBL-51)

### CODE REVIEW

File: src/main/java/org/folio/service/PasswordResetLinkServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        if (user.isPresent()) {
+          return Future.succeededFuture(user.get());
+        } else {
+          UnprocessableEntityMessage message = new UnprocessableEntityMessage("user.not.found",
+            String.format("User with id = %s in not found", pwdResetAction.getUserId()));
+          return Future.<User>failedFuture(new UnprocessableEntityException(Collections.singletonList(message)));
+        }
+      })).compose(user -> {
+        JsonObject tokenPayload = new JsonObject().put("payload",
+          new JsonObject().put("sub", user.result().getUsername()).put("user_id", user.result().getId()));
```
Comment (dmytrotur) [inline src/main/java/org/folio/service/PasswordResetLinkServiceImpl.java]: authTokenClient puts given JsonObject to property "payload", so there is no need to do it here.

File: src/main/java/org/folio/rest/impl/BLUsersAPI.java [hunk 2]
```diff
@@ -1160,4 +1174,30 @@ public void postBlUsersSettingsMyprofilePassword(UpdateCredentials entity, Map<S
...
+    passwordResetLinkService.validateLinkAndLoginUser(connectionParams)
+      .setHandler(res -> {
+        if (res.succeeded()) {
+          asyncResultHandler.handle(Future.succeededFuture(
+            PostBlUsersPasswordResetValidateResponse.respond200WithApplicationJson(res.result())));
+        } else {
+          if (res.cause().getClass() == UnprocessableEntityException.class) {
+            asyncResultHandler.handle(Future.succeededFuture(
+                PostBlUsersPasswordResetValidateResponse.respond422WithTextPlain(
+                  ExceptionHelper.handleException(res.cause()))));
```
Comment (dmytrotur) [inline src/main/java/org/folio/rest/impl/BLUsersAPI.java]: ExceptionHelper#handleException returns Response object with status and body setup, so just pass returned object to asyncResultHandler like:
} else {
          asyncResultHandler.handle(Future.succeededFuture(ExceptionHelper.handleException(res.cause())));
        }

File: NEWS.md [hunk 3]
```diff
@@ -1,3 +1,7 @@
+## 4.2.0 2018-11-23
```
Comment (kostyantyn-kh) [inline NEWS.md]: Why 4.2.0 but not 4.1.0?

File: descriptors/ModuleDescriptor-template.json [hunk 4]
```diff
@@ -4,7 +4,7 @@
   "provides": [
     {
       "id": "users-bl",
-      "version": "3.0",
+      "version": "3.2",
```
Comment (kostyantyn-kh) [inline descriptors/ModuleDescriptor-template.json]: 3.2? not 3.1?

File: ramls/mod-users-bl.raml [hunk 5]
```diff
@@ -191,3 +194,22 @@ resourceTypes:
...
                   example: "Internal server error"
+  /password-reset:
+    /validate:
+      post:
+        responses:
+          200:
+            description: "Successfully signed token"
+            body:
+              application/json:
+                type: tokenResponse
```
Comment (kostyantyn-kh) [inline ramls/mod-users-bl.raml]: Seems to me there should be 400 for Bad request

File: NEWS.md (hunk 3 above)
Comment (opochepam) [reply NEWS.md]: there is another PR for this module ([MODUSERBL-50 Implementing reset password link generation]( https://github.com/folio-org/mod-users-bl/pull/50)) and we are expecting that my changes will be merged aftere that PR so I incremented version from that PR

File: descriptors/ModuleDescriptor-template.json (hunk 4 above)
Comment (opochepam) [reply descriptors/ModuleDescriptor-template.json]: the same here

**Pull request merged**